compiler/src/main/java/com/squareup/anvil/compiler/codegen/ksp/KSAnnotationExtensions.kt

```diff
@@ -0,0 +1,75 @@
+@file:Suppress("invisible_reference", "invisible_member")
+package com.squareup.anvil.compiler.codegen.ksp
+
+import com.google.devtools.ksp.isDefault
+import com.google.devtools.ksp.symbol.KSAnnotation
+import com.google.devtools.ksp.symbol.KSClassDeclaration
+import com.google.devtools.ksp.symbol.KSType
+import com.google.devtools.ksp.symbol.KSValueArgument
+import com.squareup.anvil.compiler.internal.daggerScopeFqName
+import com.squareup.anvil.compiler.internal.mapKeyFqName
+import com.squareup.anvil.compiler.qualifierFqName
+import org.jetbrains.kotlin.name.FqName
+
+internal fun <T : KSAnnotation> List<T>.checkNoDuplicateScopeAndBoundType(
```

Comment:
Part of me is wondering if we should add some light layer of abstraction to share the larger utility extensions between ksp and embedded/reference impls, but I think this is fine for now and we can do some combining later if it there ends up being enough duplication to be worth it.